### PR TITLE
modify code with buildin function

### PR DESCRIPTION
**Pull request by Maybewuss**
## Description
reduce half memory cost and accelerate running time

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/hgt/model.py [hunk 1]
```diff
@@ -51,53 +52,43 @@ def __init__(self,
...
+                v = v_linear(h[srctype]).view(-1, self.n_heads, self.d_k)
+                q = q_linear(h[dsttype]).view(-1, self.n_heads, self.d_k)
+
+                e_id = self.edge_dict[etype]
+
+                relation_att = self.relation_att[e_id]
+                relation_pri = self.relation_pri[e_id]
+                relation_msg = self.relation_msg[e_id]
+
+                k = (k.transpose(0, 1) @ relation_att).transpose(0, 1)
```
Comment (yzh119) [inline examples/pytorch/hgt/model.py:76]: I think using PyTorch's `einsum` API will be more concise, anyways it's your freedom.

**Review by yzh119 — APPROVED** [review]
Nice contribution @Maybewuss !

File: examples/pytorch/hgt/model.py (hunk 1 above)
Comment (Maybewuss) [reply examples/pytorch/hgt/model.py:76]: Thanks for your suggestion, need I modify here ?

Comment (yzh119) [reply examples/pytorch/hgt/model.py:76]: It's up to you! Just go ahead and merge this PR when you feel it's ready.

**Pull request merged**